includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2911,6 +2912,52 @@ public function cancel_authorization( $order ) {
 		];
 	}
 
+	/**
+	 * Cancels uncaptured authorizations on order cancel.
+	 *
+	 * @param int $order_id - Order ID.
+	 */
+	public function cancel_authorizations_on_order_cancel( $order_id ) {
+		$order = new WC_Order( $order_id );
+		if ( null !== $order ) {
+			$intent_id = $this->order_service->get_intent_id_for_order( $order );
+			if ( null !== $intent_id && '' !== $intent_id ) {
+				try {
+					$request = Get_Intention::create( $intent_id );
+					$request->set_hook_args( $order );
+					$intent = $request->send();
+					$charge = $intent->get_charge();
+
+					/**
+					 * Successful but not captured Charge is an authorization
+					 * that needs to be cancelled.
+					 */
+					if ( null !== $charge
+						&& false === $charge->is_captured()
+						&& Intent_Status::SUCCEEDED === $charge->get_status()
+						&& Intent_Status::REQUIRES_CAPTURE === $intent->get_status()
+					) {
+							$request = Cancel_Intention::create( $intent_id );
+							$request->set_hook_args( $order );
+							$intent = $request->send();
+
+							$this->order_service->update_order_status_from_intent( $order, $intent );
```

Comment:
IMO, This update is unnecessary - it does too much, incl. order status updates in some scenarios. Such circular logic can break at any time.

We should keep `$this->order_service->set_intention_status_for_order( $order, $intent->get_status() );` though - webhooks will much later.

Reply:
Surfacing from another conversation branch in this PR: Having order note on success is not that important compared to the mentioned risk.

Reply:
I think it'd be prudent to have a note to indicate the change. Otherwise we will have it if authorisation is cancelled manually, but not if automatically. I'll look into extracting note creation into public methods in Order Service.

Reply:
I've extracted the note creation into a separate method. It required only a small update to the tests to accommodate changed order of the notes.